molgenis-core-ui/src/main/java/org/molgenis/ui/admin/permission/PermissionManagerController.java

```diff
@@ -44,144 +48,134 @@ public PermissionManagerController(PermissionManagerService pluginPermissionMana
+	@RequestMapping(value = "/plugin/group/{groupId}", method = GET)
 	@ResponseBody
 	public Permissions getGroupPluginPermissions(@PathVariable String groupId)
 	{
 		return pluginPermissionManagerService.getGroupPluginPermissions(groupId);
 	}
 
-	@RequestMapping(value = "/entityclass/group/{groupId}", method = RequestMethod.GET)
+	@RequestMapping(value = "/entityclass/group/{groupId}", method = GET)
 	@ResponseBody
 	public Permissions getGroupEntityClassPermissions(@PathVariable String groupId)
 	{
 		return pluginPermissionManagerService.getGroupEntityClassPermissions(groupId);
 	}
 
-	@RequestMapping(value = "/plugin/user/{userId}", method = RequestMethod.GET)
+	@RequestMapping(value = "/plugin/user/{userId}", method = GET)
 	@ResponseBody
 	public Permissions getUserPluginPermissions(@PathVariable String userId)
 	{
 		return pluginPermissionManagerService.getUserPluginPermissions(userId);
 	}
 
-	@RequestMapping(value = "/entityclass/user/{userId}", method = RequestMethod.GET)
+	@RequestMapping(value = "/entityclass/user/{userId}", method = GET)
 	@ResponseBody
 	public Permissions getUserEntityClassPermissions(@PathVariable String userId)
 	{
 		return pluginPermissionManagerService.getUserEntityClassPermissions(userId);
 	}
 
-	@RequestMapping(value = "/update/plugin/group", method = RequestMethod.POST)
+	@RequestMapping(value = "/update/plugin/group", method = POST)
 	@ResponseStatus(HttpStatus.OK)
 	public void updateGroupPluginPermissions(@RequestParam String groupId, WebRequest webRequest)
 	{
-		List<GroupAuthority> authorities = new ArrayList<GroupAuthority>();
+		List<GroupAuthority> authorities = new ArrayList<>();
 		for (MolgenisPlugin plugin : pluginPermissionManagerService.getPlugins())
 		{
-			String param = "radio-" + plugin.getId();
-			String value = webRequest.getParameter(param);
+			String pluginName = plugin.getId();
+			String value = webRequest.getParameter("radio-" + pluginName);
 
-			if (value.equalsIgnoreCase(Permission.READ.toString()) || value
-					.equalsIgnoreCase(Permission.COUNT.toString()) || value
-					.equalsIgnoreCase(Permission.WRITE.toString()) || value
-					.equalsIgnoreCase(Permission.WRITEMETA.toString()))
+			if (value.equalsIgnoreCase(COUNT.toString()) || value.equalsIgnoreCase(READ.toString()) || value
+					.equalsIgnoreCase(WRITE.toString()) || value.equalsIgnoreCase(WRITEMETA.toString()))
 			{
-				GroupAuthority authority = groupAuthorityFactory.create();
-				authority.setRole(SecurityUtils.AUTHORITY_PLUGIN_PREFIX + value.toUpperCase() + "_" + plugin.getId()
-						.toUpperCase());
-				authorities.add(authority);
+				authorities.add(createGroupAuthority(
+						AUTHORITY_PLUGIN_PREFIX + value.toUpperCase() + "_" + pluginName.toUpperCase()));
+
+				authorities.addAll(plugin.getRequiredSettingEntities().stream()
+						.map(settingEntityName -> createGroupAuthority(
+								AUTHORITY_ENTITY_READ_PREFIX + settingEntityName.toUpperCase()))
```

Comment:
I was thinking about this as well, I guess you want to supply a Map<Entity, PermissionType> for every plugin